### PR TITLE
Remove SPath

### PR DESCRIPTION
**Pull request by tobous**
Continuation of #934. Second part of refactoring, removing `SPath` entirely.

With this PR, there should be no remnants of `SPath` or its usage left. All temporary logic introduces as part of the refactoring should be removed.

## Reviewing This PR

This PR contains 5 major refactoring steps:

1. Removing all direct usages of `SPath` from the Eclipse implementation.
2. Removing all direct usages of `SPath` from the Server Implementation.
3. Removing all direct usages of `SPath` from the IntelliJ Implementation.
4. Removing all direct usages of `SPath` from the core.
5. Removing `SPath` and doing some minor cleanup.

The commits for each phase are listed below. As this is a big refactoring that required some ugly temporary solutions, I would suggest reviewing this PR by only looking at the final state (i.e. all commits combined).

Basically all of the contained changes are straight replacements of `SPath` with `IFile` or `IFolder`.

This PR also does some cleanup for changes introduced in #934.

### Step 1 - Remove usages of `SPath` from Eclipse implementation
<br>

<b>[INTERNAL][E] Remove usage of SPath from AwarenessInformationTreeElement</b>

<details><summary><b>[INTERNAL][E] Remove usage of SPath from EditorManager</b></summary>
<br>

Subsequently removes the usage of SPath from EditorAPI.

</details>

<b>[INTERNAL][E] Remove usage of SPath from EditorPool</b>

### Step 2 - Remove usages of `SPath` from Server implementation
<br>

<b>[INTERNAL][S] Remove usage of SPath from ServerEditorManager</b>

### Step 3 - Remove usages of `SPath` from IntelliJ implementation
<br>

<b>[INTERNAL][I] Remove usage of SPath from document modification handlers</b>

<b>[INTERNAL][I] Remove usage of SPath from AnnotationUpdater</b>

<b>[INTERNAL][I] Remove usage of SPath from editor pools</b>

<b>[INTERNAL][I] Remove the usage of SPath from LocalEditorHandler</b>

<b>[INTERNAL][I] Remove usage of SPath from LocalEditorManipulator</b>

<b>[INTERNAL][I] Remove usage of SPath from EditorManager</b>

<details><summary><b>[NOP][I] Remove SPath related method from VirtualFileConverter</b></summary>
<br>

Removes SPath related methods from VirtualFileConverter as they were no
longer being used.

</details>

### Step 4 - Remove usages of `SPath` from core
<br>

<b>[NOP][CORE] Remove usage of SPath from ConcurrentDocumentServer</b>

<b>[INTERNAL][CORE] Remove usage of SPath from Algorithm and Jupiter</b>

<b>[INTERNAL] Remove usage of SPath from EditorState</b>

<b>[INTERNAL] Remove usage of SPath from ISharedEditorListener</b>

<b>[INTERNAL][CORE] Remove usage of SPath from OutgoingStreamProtocol</b>

<b>[INTERNAL] Remove usage of SPath from UserEditorStateManger</b>

<b>[INTERNAL][CORE] Remove usage of SPath from instant project negotiation</b>

<b>[INTERNAL] Remove usage of SPath from IEditorManager</b>

### Step 5 - Remove `SPath` and do some minor cleanup
<br>

<details><summary><b>[JUNIT][CORE] Remove unnecessary test setup after SPath removal</b></summary>
<br>

Removes the usage of backed resource mocks that were necessary as long
as the tests included the conversion to and from SPaths.

Removes unused methods from SarosMocks.

</details>

<details><summary><b>[NOP] Remove SPath</b></summary>
<br>

Removes the class SPath as it is no longer used.

</details>

<b>[NOP][E] Remove unused mock from UndoTest</b>

<details><summary><b>[NOP] Remove remaining references to SPath</b></summary>
<br>

Removes remaining references to SPath from comments, javadoc, and method
names.

</details>

### CODE REVIEW

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           

Complexity decreasing per file
==============================
+ server/src/saros/server/editor/ServerEditorManager.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5411055&bid=17740422)

**Comment by tobous** [comment]
Rebased onto current base branch resolving minor merge issues.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           

Complexity decreasing per file
==============================
+ server/src/saros/server/editor/ServerEditorManager.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5411055&bid=17740422)

File: core/src/saros/concurrent/management/JupiterClient.java [hunk 1]
```diff
@@ -5,15 +5,14 @@
...
-import saros.activities.SPath;
 import saros.activities.TextEditActivity;
 import saros.concurrent.jupiter.Operation;
 import saros.concurrent.jupiter.TransformationException;
 import saros.concurrent.jupiter.internal.Jupiter;
 import saros.filesystem.IFile;
 import saros.session.ISarosSession;
 
-/** A JupiterClient manages Jupiter client docs for a single user with several paths */
+/** A JupiterClient manages Jupiter client docs for a single user with several file. */
```
[the remark this answers is not in the document]
Comment (tobous) [reply core/src/saros/concurrent/management/JupiterClient.java]: Done.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           

Complexity decreasing per file
==============================
+ server/src/saros/server/editor/ServerEditorManager.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5411055&bid=17740422)

**Comment by tobous** [comment]
Rebased onto current master without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 1
           

Complexity decreasing per file
==============================
+ server/src/saros/server/editor/ServerEditorManager.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5411055&bid=17740422)

**Pull request merged**